charts/trino/templates/deployment-coordinator.yaml

```diff
@@ -82,14 +82,38 @@ spec:
             - name: http
               containerPort: {{ .Values.service.port }}
               protocol: TCP
+          {{- if .Values.coordinator.livenessProbe.enabled }}
           livenessProbe:
             httpGet:
               path: /v1/info
               port: http
+            initialDelaySeconds: {{ .Values.coordinator.livenessProbe.initialDelaySeconds }}
+            periodSeconds: {{ .Values.coordinator.livenessProbe.periodSeconds }}
+            timeoutSeconds: {{ .Values.coordinator.livenessProbe.timeoutSeconds }}
+            successThreshold: {{ .Values.coordinator.livenessProbe.successThreshold }}
+            failureThreshold: {{ .Values.coordinator.livenessProbe.failureThreshold }}
+          {{- else if .Values.coordinator.customLivenessProbe }}
+          {{- with .Values.coordinator.customLivenessProbe }}
+          livenessProbe:
+            {{- toYaml . | nindent 12 }}
+          {{- end }}
+          {{- end }}
```

Comment:
Maybe let's split this by moving the default definition into helpers.tpl and leaving values.yaml empty and have the deployment pick the default if values is empty.

that way it keeps the chart simple by providing good defaults and allows anyone who wants customisation to provide their own `livenessProbe` node.

Same for readinessProbe.

Reply:
I don't see the need for using `_helpers.tpl`. Instead, why not use the [helm `default` function?](https://helm.sh/docs/chart_template_guide/functions_and_pipelines/#using-the-default-function)
So it would look like:
```yaml
initialDelaySeconds: {{ .Values.coordinator.readinessProbe.initialDelaySeconds | default 20 }}
periodSeconds: {{ .Values.coordinator.readinessProbe.periodSeconds | default 10 }}
timeoutSeconds: {{ .Values.coordinator.readinessProbe.timeoutSeconds | default 5 }}
successThreshold: {{ .Values.coordinator.readinessProbe.successThreshold | default 6 }}
failureThreshold: {{ .Values.coordinator.readinessProbe.failureThreshold | default 1 }}
```

If this is the case, do you also want `values.yaml` to look like this?
```yaml
livenessProbe: {}
  # enabled: true
  # initialDelaySeconds: 20
  # periodSeconds: 10
  # timeoutSeconds: 5
  # failureThreshold: 6
  # successThreshold: 1
readinessProbe: {}
  # enabled: true
  # initialDelaySeconds: 20
  # periodSeconds: 10
  # timeoutSeconds: 5
  # failureThreshold: 6
  # successThreshold: 1
customLivenessProbe: {}
customReadinessProbe: {}
```

Reply:
I'd say we can remove `customLivenessProbe` and just use one of them. Keeps things simpler (for consumers of the chart for sure, maybe not for authors which is fine).

Reply:
Ok. I've removed `customReadinessProbe`, `customLivenessProbe`, `readinessProbe.enabled`, `livenessProbe.enabled` as well
All of the `if` statements are gone and it will use the defaults if the selected values don't exist.